cadquery/occ_impl/shapes.py

```diff
@@ -730,6 +730,47 @@ def makeCircle(cls, radius, pnt=Vector(0, 0, 0), dir=Vector(0, 0, 1), angle1=360
                                              True).Value()
             return cls(BRepBuilderAPI_MakeEdge(circle_geom).Edge())
 
+    @classmethod
+    def makeEllipse(cls, x_radius, y_radius, pnt=Vector(0, 0, 0), dir=Vector(0, 0, 1),
+                    angle1=360.0, angle2=360.0):
+        """
+        Interpolate a spline through the provided points.
+        :param cls:
+        :param x_radius: x radius of the ellipse (along the x-axis of plane the ellipse should lie in)
+        :param y_radius: y radius of the ellipse (along the y-axis of plane the ellipse should lie in)
+        :param pnt: vector representing the center of the ellipse
+        :param dir: vector representing the direction of the plane the ellipse should lie in
+        :param angle1: start angle of arc
+        :param angle2: end angle of arc (angle2 == angle1 return closed ellipse = default)
+        :return: an Edge
+        """
+
+        pnt = Vector(pnt).toPnt()
+        dir = Vector(dir).toDir()
+
+        ax1 = gp_Ax1(pnt, dir)
+        ax2 = gp_Ax2(pnt, dir)
+        
+        if y_radius > x_radius:
+            # swap x and y radius and rotate by 90° afterwards to create an ellipse with x_radius < y_radius
+            correction_angle = 90.0 * DEG2RAD
+            ellipse_gp = gp_Elips(ax2, y_radius, x_radius).Rotated(ax1, correction_angle)
+        else:
+            correction_angle = 0.0
+            ellipse_gp = gp_Elips(ax2, x_radius, y_radius)
+        
+        if angle1 == angle2:  # full ellipse case
```

Comment:
Do you really need this if else? Wouldn't GC_MakeArcOfEllipse suffice?

Reply:
makeCircle is built the same way, and I wanted to be as similar with my ellipse implementation as possible.

Reply:
see also https://www.opencascade.com/doc/occt-6.9.0/refman/html/class_g_c___make_arc_of_ellipse.html 
`GC_MakeArcOfEllipse` constructor requires a `gp_Elips`